fetch.bs

```diff
@@ -2006,6 +1996,34 @@ is a <a>filtered response</a> whose
 <p>A <dfn export id=concept-stale-response>stale response</dfn> is a <a for=/>response</a> that is
 not a <a>fresh response</a> or a <a>stale-while-revalidate response</a>.
 
+<hr>
+
+<p>To <dfn export id=concept-response-location-url>extract a location URL</dfn>, given a
+<a for=/>response</a> <var>response</var>:
+
+<ol>
+ <li><p>If <var>response</var>'s <a for=response>status</a> is not a <a>redirect status</a>, then
+ return null.
+
+ <li><p>Let <var>location</var> be the result of <a>extracting header list values</a> given
+ `<code>Location</code>` and <var>response</var>'s <a for=response>header list</a>.
+ <!-- https://github.com/whatwg/fetch/issues/814#issuecomment-431366126 -->
+
+ <li>
+  <p>If <var>location</var> is a <a for=header>value</a>, then set <var>location</var> to the result
+  of <a lt="url parser">parsing</a> <var>location</var> with <var>response</var>'s
+  <a for=response>URL</a>.
```

Comment:
The URL parser takes an optional base URL, not a URL-or-null. So it'd be most correct to state this as something like "parsing _location_, with _response_'s URL if _response_'s URL is not null."

Reply:
I plan to update the URL Standard to make the default more clear now that we have Infra language for it. I guess I might as well try to land that before this.

Reply:
https://github.com/whatwg/url/pull/575